swarm/storage/dpa_test.go

```diff
@@ -27,14 +27,14 @@ import (
 const testDataSize = 0x1000000
 
 func TestDPArandom(t *testing.T) {
-	tdb, err := newTestDbStore(false)
+	tdb, err := newTestDbStore(false, false)
```

Comment:
I rather like named boolean variables than primitives, easier to understand the code (but more verbose, so up to you)

Reply:
I agree with this. We should rethink the way we create constructors, because at the moment it is very difficult to reason about such code, without knowing by heart all these parameters.

Reply:
For APIs I can agree with this. For internal (test) functions, I think it's overkill.

Reply:
I disagree that it is an overkill. I think most of our tests are quite difficult to reason about because the setup phase includes so many constructors with nils and booleans, that it is very mentally challenging to track what is going on. The `swarm/storage` package is particularly bad in this regard - a lot of indirection, almost no comments on the tests and different functions triggering methods with a variety of combinations (nil, not nil, true, false, etc.).

I am not saying this is in-scope of this PR, but this is definitely a big problem in our test suite.